debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java

```diff
@@ -108,9 +109,19 @@ private Object convertToLocalDate(Column column, String value) {
      * @return the converted value;
      */
     private Object convertToLocalDateTime(Column column, String value) {
-        if (ALL_ZERO_TIMESTAMP.equals(value) && column.isOptional()) return null;
-        if (ALL_ZERO_TIMESTAMP.equals(value)) value = EPOCH_TIMESTAMP;
-        String timestampFormat = timestampFormat(column.length());
+        final boolean matches = ALL_ZERO_TIMESTAMP.matcher(value).matches();
+        if (matches && column.isOptional()) return null;
+        if (matches) {
+            value = EPOCH_TIMESTAMP;
+            if (column.length() > 0) {
+                final StringBuilder sb = new StringBuilder(EPOCH_TIMESTAMP).append('.');
```

Comment:
Why is it that the 0 are appended? A comment would be good. Also why are length 0's appended?